### ISSUE TITLE
@nstudio/nativescript-loading-indicator: recent iOS view changes causes indicator to display behind modals

### ISSUE THREAD

**Issue by darklight365** [nstudio/nativescript-plugins#135]
The recent release of the @nstudio/nativescript-loading-indicator plugin will now render the loading indicator behind the current modal being displayed (when using the Angular ModalDialogService to display a modal).

In order to correct this the view would need to be passed in as an ios option, for example:
`
    const options: OptionsCommon = {dimBackground: true};
    if (__IOS__) {
      options.ios = {
        view: UIApplication.sharedApplication.windows[0]
      }
    }
`

Is this the suggested/recommended way to handle this with the recent changes made for _getRootWindow or is there a better approach?

A reproducible stackblitz can be found here:
https://stackblitz.com/edit/nativescript-angular14-modal-page-router-issue-mjsbw4?file=readme.md

**Comment by NathanWalker** [comment]
Hi @darklight365 please use `NativeDialogService` for all modal needs. `ModalDialogService` should really no longer be used and likely to be deprecated on next major and removed in the future. If you have any issue with `NativeDialogService` please let us know.

I updated your StackBlitz with few dependencies that were needed and confirmed that `NativeDialogService` works as expected:
https://stackblitz.com/edit/nativescript-angular14-modal-page-router-issue-5tqsli?file=src%2Fapp%2Fitem%2Fitems.component.ts

**Comment by darklight365** [comment]
Thanks @NathanWalker for checking this out.

I figured that would be the suggested correction (which was why I had added it as an option) but just wanted to bring this to your attention in case you wanted to correct the plugin to support people that may not have the ability/bandwidth to update (for whatever reason) their code.  I also wanted to document the OptionsCommon workaround (adding ios view)  in case others would need it in the future.

Another quick fix would be to pin the @nstudio/nativescript-loading-indicator dependency to 4.1.2 within their project as well.

Thanks again for your time!
